actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1168,13 +1168,27 @@ module Resources
         CANONICAL_ACTIONS = %w(index create new show update destroy)
 
         class Resource # :nodoc:
+          class << self
+            def default_actions(api_only)
+              if api_only
+                [:index, :create, :show, :update, :destroy]
+              else
+                [:index, :create, :new, :show, :update, :destroy, :edit]
+              end
+            end
+          end
+
           attr_reader :controller, :path, :param
 
           def initialize(entities, api_only, shallow, options = {})
             if options[:param].to_s.include?(":")
               raise ArgumentError, ":param option can't contain colons"
             end
 
+            if (invalid_actions = (options.values_at(:only, :except).flatten.compact - default_actions)).any?
+              raise ArgumentError, "expected :only and :except to be one or more of #{default_actions}, got #{invalid_actions}"
+            end
```

Comment:
The method `assert_valid_keys` would be a better approach as it's used elsewhere within Rails to perform the same task, e.g.

https://github.com/rails/rails/blob/778039b896be1dcb3da606f761921d006cab67c9/actionview/lib/action_view/helpers/output_safety_helper.rb#L43-L50

Reply:
Good shout, I'll address.

Reply:
I don't think it works here since we're asserting the values and not the keys.